[11.0] [MIG] report_qweb_pdf_watermark

schout-it opened the pull request:
migration, now with module history

hbrunn committed:
report_qweb_pdf_watermark (#86)

oca-transbot committed:
OCA Transbot updated translations from Transifex

StefanRijnhart committed:
[MIG] report_qweb_pdf_watermark

StefanRijnhart committed:
[ADD] Prioritize background transparency

oca-transbot committed:
OCA Transbot updated translations from Transifex

hbrunn committed:
[IMP] use PyPDF2 if possible

etobella committed:
[FIX] Fix Watermark from PNG

oca-transbot committed:
OCA Transbot updated translations from Transifex

hbrunn committed:
[UPD] build environment

JordiBForgeFlow committed:
fix test_report_qweb_signer - tests were causing travis to stall

StefanRijnhart committed:
[FIX] Don't break printing some upstream reports with invalid models
When no background configuration is present on the report definition,
don't try to browse the report model especially because this breaks
some upstream reports (see https://github.com/odoo/odoo/pull/23389)

oca-transbot committed:
OCA Transbot updated translations from Transifex

oca-travis committed:
[UPD] Update report_qweb_pdf_watermark.pot

schout-it committed:
[MIG] report_qweb_pdf_watermark: Migration to 11.0

schout-it commented:
Had a look at the test errors, but it seems that when the test cursor is used, render_qweb_pdf returns the html version of the report.
Which renders the test useless, as _run_wkhtmltopdf is never called

pedrobaeza commented:
If you look at the code:

https://github.com/odoo/odoo/blob/750e70e0a12c11d50998a44dff447f07fd7d06c3/odoo/addons/base/ir/ir_actions_report.py#L566-L567

what is preventing the PDF generation is the absence of a `test_report_directory` configuration variable.

@hbrunn I remember you trick this variable in one .travis.yml file, isn't it? Can you point us how to do it?

schout-it commented:
That is one part, but a little further there is a check for the cursor type which will also block:
https://github.com/odoo/odoo/blob/750e70e0a12c11d50998a44dff447f07fd7d06c3/odoo/addons/base/ir/ir_actions_report.py#L593-L594

pedrobaeza commented:
Ah, that new one is good for preventing deadlocks as said, but then I don't know how to do it...

pedrobaeza commented:
I have found how to do it (at least the directory part): https://github.com/OCA/edi/blob/11.0/.travis.yml#L42

Please try.

pedrobaeza commented:
Now maybe you should add also wkhtmltopdf: https://github.com/OCA/edi/blob/11.0/.travis.yml#L26

schout-it commented:
wkhtml2pdf added, and also added pip dependency for PyPDF2

pedrobaeza reviewed (approved):
Squashing the 11.0 commit history before merging.

LoisRForgeFlow commented on report_qweb_pdf_watermark/README.rst:
```diff
@@ -0,0 +1,73 @@
+
+.. image:: https://odoo-community.org/website/image/ir.attachment/5784_f2813bd/datas
+    :alt: Try me on Runbot
+    :target: https://runbot.odoo-community.org/runbot/143/8.0
```
11.0

schout-it replied:
seems even with 11.0 I can't test the module.
What should I change the 143 to, to get the right runbot?

schout-it replied:
answering my own question, as long as this is not merged, it will not show up on that runbot

pedrobaeza commented:
Now you can try on runbot

ovnicraft commented:
@schout-it please resolve conflicts

MiquelRForgeFlow commented:
You should do a correct rebase.

schout-it commented:
had some trouble rebasing, did a diff at the end and fixed what was still wrong

MiquelRForgeFlow commented:
please do
```
    git fetch [remote] 11.0
    git rebase -i [remote]/11.0
```

where `[remote]` should be `origin` (if `origin` points to OCA repo) or `OCA` or `other_name_you_use_for_OCA_remote`

schout-it commented:
That is what I did, and after resolving all the rebase conflicts I did a diff to be sure and saw the 2 changes that where not picked up in the rebase

Did it again to pick up the latest changes

MiquelRForgeFlow commented:
It seems you are not fetching correctly for some reason.

please do:
```
    git remote update
```
and then the rebase

schout-it commented:
before trying the last time, I deleted the local repo, cloned it back and added oca as a remote.
The rebase always starts back at the beginning
on everythin not related to the changes I made, I do a 
git checkout --ours [path of the file]

It seems some patches will not apply, will need to look at them more closely

khaeusler commented on report_qweb_pdf_watermark/models/report.py:
```diff
@@ -1,43 +1,54 @@
-    from PyPDF2.utils import PdfReadError  # pylint: disable=W0404
-except ImportError:
-    pass
+from PyPDF2 import PdfFileWriter, PdfFileReader  # pylint: disable=W0404
```
the pylint flag for reimported is not necessary

pedrobaeza replied:
Please don't remove the import protection

pedrobaeza commented:
Can you squash a bit commit history for merging this?

MiquelRForgeFlow commented:
I don't understand why commits from other modules are here.

pedrobaeza commented:
Trying to rebase myself.
